src/dispatch/plugins/dispatch_slack/incident/interactive.py

```diff
@@ -2140,3 +2166,73 @@ def monitor_link_button_click(
         if enabled is True
         else f"This monitor is now ignored. Dispatch won't remind this incident channel about it again.\n\n *Weblink:* {button.weblink}"
     )
+
+
+@app.action(
+    TaskNotificationActionIds.update_status,
+    middleware=[button_context_middleware, db_middleware],
+)
+def handle_update_task_status_button_click(
+    ack: Ack,
+    body: dict,
+    client: WebClient,
+    context: BoltContext,
+    db_session: Session,
+):
+    """Handles the update task button in the list-my-tasks message."""
+    ack()
+
+    button = TaskMetadata.parse_raw(body["actions"][0]["value"])
+
+    resolve = True
+    if button.action_type == "reopen":
+        resolve = False
+
+    # When the task originated from the Dispatch front-end.
+    # Tasks created in the front-end do not have an associated "resource".
+    # Thus, no "resource_id" entry is necessary.
+    from_drive = True if button.resource_id is not None else False
+
+    if from_drive:
+        # When the task originated from the drive plugin
+        task = task_service.get_by_resource_id(
+            db_session=db_session, resource_id=button.resource_id
+        )
+    else:
+        task = task_service.get(db_session=db_session, task_id=button.task_id)
+
+    # avoid external calls if we are already in the desired state
+    if resolve and task.status == TaskStatus.resolved:
+        return
+
+    if not resolve and task.status == TaskStatus.open:
+        return
+
+    if from_drive:
+        # we don't currently have a good way to get the correct file_id (we don't store a task <-> relationship)
+        # lets try in both the incident doc and PIR doc
+        drive_task_plugin = plugin_service.get_active_instance(
+            db_session=db_session, project_id=task.incident.project.id, plugin_type="task"
+        )
+        try:
+            file_id = task.incident.incident_document.resource_id
+            drive_task_plugin.instance.update(file_id, button.resource_id, resolved=resolve)
+        except DispatchException:
+            file_id = task.incident.incident_review_document.resource_id
+            drive_task_plugin.instance.update(file_id, button.resource_id, resolved=resolve)
+    else:
+        task_service.resolve_or_reopen(
+            db_session=db_session, task_id=task.id, status=TaskStatus.resolved
+        ) if resolve is True else task_service.resolve_or_reopen(
+            db_session=db_session, task_id=task.id, status=TaskStatus.open
+        )
+
+    draw_task_modal(
+        caller_only=None,
```

Comment:
I think we need to retain this value across clicks so that we draw the correct tasks.